chore(docs): replace typefaces with fontsource (#27313)
gatsbyjs/gatsby: docs/docs/recipes/styling-css.md

- A [Gatsby site](/docs/quick-start)
- The [Gatsby CLI](/docs/gatsby-cli/) installed
- A chosen font package from [Fontsource](https://github.com/fontsource/fontsource)

### Directions

This example shows how to set up the [Open Sans](https://fonts.google.com/specimen/Open+Sans) font. If you have a different Google Font you want to use, you can find the corresponding package in [NPM](https://www.npmjs.com/search?q=fontsource) or the [packages directory in the Fontsource repository](https://github.com/fontsource/fontsource/tree/master/packages).

1. Run `npm install fontsource-open-sans` to download the necessary package files.

> 2. Then within your app entry file or site component, import it in. In Gatsby, it is recommended you import it via the layout template (`layout.js`), however, importing via page component (`index.js`), or `gatsby-browser.js` are viable alternatives.

Author: > https://github.com/gatsbyjs/gatsby/pull/27313#discussion_r514596462

Would this fit the bill to highlight a recommended method over the previous passive approach?

Reviewer: Making some copy edits, but other than that it looks good! Thanks for the update!
```suggestion
2. Then within your app entry file or site component, import the font package. It is recommended you import it via the layout template (`layout.js`). However, importing via page component (`index.js`), or `gatsby-browser.js` are viable alternatives.
```

- A [Gatsby site](/docs/quick-start)
- The [Gatsby CLI](/docs/gatsby-cli/) installed
- Choosing a font package from [Fontsource](https://github.com/fontsource/fontsource)

### Directions

This is an installation example using Open Sans, applicable to all other fonts searchable via [NPM](https://www.npmjs.com/) or the [packages](https://github.com/fontsource/fontsource/tree/master/packages) directory in the repository.

1. Run `npm install fontsource-open-sans` to download the necessary package files. If you wish to use a different typeface, search up `fontsource-<insert font name>` on [NPM](https://www.npmjs.com/) for specific instructions tailored to that font.

> 2. Then within your app entry file or site component, import it in. In Gatsby, you could choose to import it into a layout template (`layout.js`), page component (`index.js`), or `gatsby-browser.js`.

Reviewer: I would change this to add a stronger recommendation on where to add the import. We want this guide to be easy to follow for new folks who might not have enough context to pick one of the options for themselves.

You can list the other options afterward, so that they know they exist. But pick one to be the recommended approach.

```jsx:title=src/components/layout.js
import "fontsource-open-sans" // Defaults to weight 400 with all styles included.

> import "fontsource-open-sans/500.css" // Weight 500 with all styles included.

Reviewer: Can you add in some context for these second two import examples?

For the last line in particular, I'm not sure what the comment means.

Author: This one is was a little bit of a challenge to keep concise but informative enough.
Each import example shows possible ways to import different weights and styles.

- First one is the default import which is weight 400 that includes normal and italic styles.
- The second one aims to show that you can specify weights, such as 500, by changing the import path. This still includes styles normal and italic.
- The third one shows the ability to specify the style normal or italics by changing the import path yet again. i.e. "fontsource-open-sans/600-italic.css" as an alternative to the example.

Reviewer: Got it. So they don't have to do all three of these imports, they can choose whichever one they want to get the exact styling they need.

It might be helpful to split out the second two to be their own code block.

So you'd have the first one, showing a generic import. And then you'd have a text note saying something like "If you only want to import a particular font weight, you can also import a single file from the font package." and then have the second two examples.

That way the inline comments don't have to change but there's at least some context beforehand.

Author: I made the appropriate changes.

import "fontsource-open-sans/900-normal.css" // Select either normal or italic.
```

3. Once it's imported, you can reference the font name in a CSS stylesheet, CSS Module, or CSS-in-JS.

```css:title=src/components/layout.css
body {
  font-family: "Open Sans";
}
```

> _NOTE: The range of supported weights and styles a font may support is shown on each package README file._

Reviewer: I would add this note up above to the place where you're showing the different examples of font weights. (Line 409)

> _NOTE: The range of supported weights and styles a font may support is shown on each package README file._

Reviewer: Tweaking this styling to make it more consistent.

```suggestion
**Note**: The range of supported weights and styles a font may support is shown in each package's README file.
```

### Additional resources

> - [Fontsource](https://github.com/fontsource/fontsource) - Main monorepo
> - [[WIP] Search Directory](https://fontsource.github.io/search-directory/) - Currently open to contributions improving the user experience
> - [Typography.js](/docs/typography-js/) - Another option for using Google fonts on a Gatsby site

Reviewer: For the sake of making this easier to maintain, I wouldn't add this WIP link here if it's not a finished resource yet.

```suggestion
- [Fontsource repo on GitHub](https://github.com/fontsource/fontsource)
- [Typography.js](/docs/typography-js/) - Another option for using Google fonts on a Gatsby site
```
